### PR TITLE
release: add homebrew release job

### PR DESCRIPTION
**Pull request by chenrui333**
add homebrew release job to automate the release bump task.

TODO:
- need a committer token
- need a homebrew-core fork for openfga org

cc @aaguiarz

### CODE REVIEW

**Comment by miparnisari** [comment]
Thanks @chenrui333! I created HOMEBREW_COMMITTER_TOKEN and saved it as a repository secret. Expires on Thu, Mar 6 2025. FYI @openfga/backend.

**Review by rhamzeh — DISMISSED** [review]

**Review by rhamzeh — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -195,3 +195,15 @@ jobs:
...
+  homebrew:
+    needs: [ goreleaser ]
+    name: Bump Homebrew formula
+    runs-on: ubuntu-latest
+    steps:
+      - uses: mislav/bump-homebrew-formula-action@b3327118b2153c82da63fd9cbf58942146ee99f0 # v3.1
+        with:
+          formula-name: openfga
+        env:
+          # token belongs to @miparnisari and expires on March 6, 2025.
```
Comment (willvedd) [inline .github/workflows/release.yaml:208]: We need a SA for this. No personal tokens.

**Review by willvedd — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (chenrui333) [reply .github/workflows/release.yaml:208]: but this probably can be done in another PR?

Comment (miparnisari) [reply .github/workflows/release.yaml:208]: I'll try to create a service account for OpenFGA so we can avoid using personal tokens. Hang tight...

**Review by willvedd — APPROVED** [review]

**Pull request merged**